plugin/src/main/java/org/opensearch/ml/action/connector/UpdateConnectorTransportAction.java

```diff
@@ -93,6 +95,11 @@ protected void doExecute(Task task, ActionRequest request, ActionListener<Update
                 if (Boolean.TRUE.equals(hasPermission)) {
                     connector.update(mlUpdateConnectorAction.getUpdateContent(), mlEngine::encrypt);
                     connector.validateConnectorURL(trustedConnectorEndpointsRegex);
+
+                    if (connector instanceof HttpConnector && ((HttpConnector) connector).getLastUpdateTime() != null) {
```

Comment:
Why only HttpConnector? Can't we just check `connector.getLastUpdateTime() != null`

Reply:
I didn't want to bloat the [connector interface](https://github.com/opensearch-project/ml-commons/blob/main/common/src/main/java/org/opensearch/ml/common/connector/Connector.java) with a getter as it already has two getters. Also AbstractConnector has the time fields not the connector interface

Reply:
what if `getLastUpdateTime` is null? Don't we want to update those values also?

Reply:
what's the concern if we just add `connector.setLastUpdateTime(Instant.now());` without any checks?

Reply:
IMO the updates should be built-in (but overridable) defaults on the AbstractConnector.  May not need a setter for it as it should generally done automatically.

Reply:
Hi @dhrubo-os, @dbwiddis  

This method will update connectors when it does this my theory is that deserializes the connector from an index and constructs the HTTPConnector object when this happens it is very hard to control the update fields (on instantiation).

If we decide to add the time fields in the abstract Connector constructor we have a problem between old connectors and new ones. This results in a unexpected bug when connectors that didn't have these fields get brought to life; every subsequent update on the connector makes the time fields a new number every time.

Now if we wanted old connectors to not have the time fields (what our users expect when they update a old connector) they wont see the time fields because this method checks if an update should work on time only if they have the field. When a connector thats been created with the code change, they will see time fields and will update accordingly well.

Please let me know if you are still confused I would be happy to explain more in depth :)